File: measures.py
from globals import SIM_TIME, LOOPS

# Measures
# Time (days) until placement - measure no.1
tpa_lst = [{j: [0 for z in range(0, LOOPS)] for j in range(0, SIM_TIME + 1)} for i in range(6)]

# Time in center - measure no.2
tic_avg = [{j: {j: 0 for j in range(1, SIM_TIME+1)} for j in range(0, LOOPS)},
           {j: {j: 0 for j in range(1, SIM_TIME+1)} for j in range(0, LOOPS)},
           {j: {j: 0 for j in range(1, SIM_TIME+1)} for j in range(0, LOOPS)}]

# Expectation of number of packages which returned to the logistics center - measure no. 3: not relevant here
pack_return = [0 for j in range(0, LOOPS)]

# Expectation of number of customers that needed to return the day after - measure no. 4
customer_return = [0 for j in range(0, LOOPS)]

File: Measures.js
import { loops, simTime } from './Globals.js';

const zeroes = new Array(loops).fill(0);
const zeroesArray = new Array(simTime + 1).fill(zeroes);

const tpaZeroesObject = {}
for (let i = 0; i < simTime + 1; i++) {
  tpaZeroesObject[i] = zeroesArray[i]
}

export const tpaList = new Array(6).fill(tpaZeroesObject);

const ticZeroesObject = {}
for (let i = 0; i < simTime + 1; i++) {
  ticZeroesObject[i] = 0
}
const ticObjectsObject = {}
for (let i = 0; i < simTime + 1; i++) {
  ticObjectsObject[i] = ticZeroesObject;
}

export const ticAverage = new Array(3).fill(ticObjectsObject)


export const packReturn = new Array(loops).fill(0)
export const costReturn = new Array(loops).fill(0)